src/vsr/replica.zig

```diff
@@ -3926,43 +3986,36 @@ pub fn ReplicaType(
             }
 
             // Request any missing or disconnected headers:
-            if (self.commit_min != self.op) {
-                var broken = self.journal.find_latest_headers_break_between(
-                    self.commit_min + 1,
-                    self.op,
+            if (self.journal.find_latest_headers_break_between(
+                self.op_repair_min(),
+                self.op,
+            )) |range| {
+                log.debug(
+                    "{}: repair: break: view={} break={}..{} (commit={}..{} op={})",
+                    .{
+                        self.replica,
+                        self.view,
+                        range.op_min,
+                        range.op_max,
+                        self.commit_min,
+                        self.commit_max,
+                        self.op,
+                    },
                 );
-                if (broken) |range| {
-                    log.debug(
-                        "{}: repair: break: view={} op_min={} op_max={} (commit={}..{} op={})",
-                        .{
-                            self.replica,
-                            self.view,
-                            range.op_min,
-                            range.op_max,
-                            self.commit_min,
-                            self.commit_max,
-                            self.op,
-                        },
-                    );
-                    assert(range.op_min > self.commit_min);
-                    assert(range.op_max < self.op);
-                    // A range of `op_min=0` or `op_max=0` should be impossible as a header break:
-                    // This is the root op that is prepared when the cluster is initialized.
-                    assert(range.op_min > 0);
-                    assert(range.op_max > 0);
-
-                    if (self.choose_any_other_replica()) |replica| {
-                        self.send_header_to_replica(replica, .{
-                            .command = .request_headers,
-                            .cluster = self.cluster,
-                            .replica = self.replica,
-                            .view = self.view,
-                            .commit = range.op_min,
-                            .op = range.op_max,
-                        });
-                    }
-                    return;
+                assert(range.op_min >= self.op_repair_min());
+                assert(range.op_max < self.op);
+
+                if (self.choose_any_other_replica()) |replica| {
```

Comment:
What happens if there's no other replica? It feels a bit weird that wee do nothing for that case --- feels like we should maybe shutdown here if sole_replica() ?

Reply:
:+1: I think this can be `self.choose_any_other_replica().?` since solo replicas cannot tolerate corruption anyhow.

Reply:
Hm, I think the same would be true for another call to this function, so perhaps it can just assert(!sole_replica()) rather than return a null?